src/os/bluestore/BlueFS.cc

```diff
@@ -658,15 +660,25 @@ int BlueFS::_replay(bool noop, bool to_stdout)
       }
     }
     if (uuid != super.uuid) {
-      dout(10) << __func__ << " 0x" << std::hex << pos << std::dec
-               << ": stop: uuid " << uuid << " != super.uuid " << super.uuid
-               << dendl;
+      if (seen_recs) {
+	dout(10) << __func__ << " 0x" << std::hex << pos << std::dec
+		 << ": stop: uuid " << uuid << " != super.uuid " << super.uuid
+		 << dendl;
+      } else {
+	derr << __func__ << " 0x" << std::hex << pos << std::dec
+		 << ": stop: uuid " << uuid << " != super.uuid " << super.uuid
+		 << ", block dump: \n";
+	bufferlist t;
+	t.substr_of(bl, 0, super.block_size);
+	t.hexdump(*_dout);
+	*_dout << dendl;
+      }
       break;
     }
     if (seq != log_seq + 1) {
-      dout(10) << __func__ << " 0x" << std::hex << pos << std::dec
-               << ": stop: seq " << seq << " != expected " << log_seq + 1
-               << dendl;
+      derr << __func__ << " 0x" << std::hex << pos << std::dec
+           << ": stop: seq " << seq << " != expected " << log_seq + 1
+           << dendl;
```

Comment:
I think this may generate some log noise, since the bluefs log may end up overwriting older log content, and this will mark the end of replay for the valid portion of the bluefs log.  That may be confusing to someone reading the debug log...

Reply:
Maybe this should be dependent on seen_recs too?